content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -504,6 +504,22 @@ stops adjusting the target (and sets the `ScalingActive` Condition on itself
 to `false`) until you reactivate it by manually adjusting the target's desired
 replica count or HPA's minimum replica count.
 
+### Migrating Deployments and StatefulSets to horizontal autoscaling
+
+When an HPA is enabled, it is recommended that the value of `spec.replicas` of
+the Deployment and/or Statefulset object is removed.  Without doing so, anytime
+a change to that object is applied, example via `kubectl apply -f
+deployment.yaml`, will instruct Kubernetes to scale the current number of Pods
+to the desired value retained in the `spec.replicas` key. This may not be
+desired and could be troublesome when an HPA is active.
+
+Keep in mind that the removal of `spec.replicas` will incur a one time
+degradation of Pod counts as the default value of this key is 1 (reference
```

Comment:
As @apelisse explains on the related issue, this degradation only occurs when using client-side apply. There is also a workaround that is probably worth mentioning in these docs, given the consequences of scaling down to 1 even temporarily for a production workload: https://github.com/kubernetes/kubernetes/issues/25238#issuecomment-406415297.

Reply:
Ahh, so I thought this was a proposed fix from way back when for the issue at hand.  I've confirmed that solution does indeed work today.  I've also added a tidbit on serverside applying of files.  Because it's a flag that blocks the applying of a configuration I don't want to dive too much into the details of what should be done as this starts to stray away from the overall guide in general.

Let me know what you think!

Reply:
I'm overall in favor of documenting the “how to turn on (horizontal) autoscaling for a workload” part - this page _is_ a task.

What's actually missing is a concept or reference page about autoscaling. Lots of what's in this existing page should move out of the task page towards that missing concept / reference page.

It's OK _not_ to do the big refactor (that's needed!) in **this** PR.